dtool/src/dtoolutil/executionEnvironment.cxx

```diff
@@ -62,7 +62,7 @@ extern char **environ;
 #include <sys/sysctl.h>
 #endif
 
-#if defined(IS_LINUX) || defined(IS_FREEBSD)
+#if (defined(IS_LINUX) || defined(IS_FREEBSD)) && !defined(__wasi__)
```

Comment:
Neither IS_LINUX nor IS_FREEBSD should be set for WASI to begin with

Reply:
That's just temporary later it should be
defined(IS_LINUX) || defined(IS_FREEBSD) || defined(__wasi__)
because dlfcn is on the way ( though it's via static linking for now )